src/Frame.cc

```diff
@@ -1276,8 +1276,20 @@ bool Frame::GetRegionData(const casacore::LattRegionHolder& region, std::vector<
         casacore::Slicer slicer(start, count); // entire subimage
         std::unique_lock<std::mutex> ulock(_image_mutex);
         sub_image.doGetSlice(tmp, slicer);
+
+        // Get mask that defines region in subimage bounding box
+        casacore::Array<bool> tmpmask;
+        sub_image.doGetMaskSlice(tmpmask, slicer);
         ulock.unlock();
 
+        // Apply mask to data
+        std::vector<bool> datamask = tmpmask.tovector();
```

Comment:
I assume that this is to get a 1D mask to match the shape of `data`. I *think* from the casacore code that this doesn't unnecessarily copy data if the storage is contiguous. But could we not access `data` here through the 2D `tmp` array instead, matching the shape of the mask?

Reply:
I would have to use casacore::ArrayIterator, which according to the perflog output adds ~50 ms per region histogram.  But if copying is necessary for contiguous storage it would copy a 'cursor' at a time instead of the entire array.

Reply:
I pushed branch pam/689_histogram_iter

Reply:
If the array iteration is less efficient, the current approach is probably fine -- I was concerned about possible copying of data when the region is very large.